File: edtc_python/testsrs.py
# Filename: testsrs.py
# Author: John Stachurski
# Date: December 2008
# Corresponds to: Listing 6.2

from srs import SRS               # Import from listing 6.1
from random import lognormvariate 

alpha, sigma, s, delta = 0.5, 0.2, 0.5, 0.1  
# Define F(k, z) = s k^alpha z + (1 - delta) k
F = lambda k, z: s * (k**alpha) * z + (1 - delta) * k 
lognorm = lambda: lognormvariate(0, sigma) 

solow_srs = SRS(F=F, phi=lognorm, X=1.0)
P1 = solow_srs.sample_path(500)   # Generate path from X = 1
solow_srs.X = 60                  # Reset the current state 
P2 = solow_srs.sample_path(500)   # Generate path from X = 60

File: edtc_matlab/testsrs.m
% Filename: testsrs.m
% Author: Andy Qi
% Date: December 2008
% Corresponds to: Listing 6.2

alpha = 0.5;
sigma = 0.2;
s = 0.5;
delta = 0.1;
% Define F(k, z) = s * k ^ alpha * z + (1 - delta ) * k
F = @(k,z) s * (k^alpha) * z + (1 - delta) * k;
lognorm = @() lognrnd(0,sigma);

solow_srs = srs(F,lognorm,1.0);
P1 = sample_path(solow_srs,500); % Generate Path from X = 1
solow_srs.X = 60;                % Reset the current state
P2 = sample_path(solow_srs,500); % Generate Path from X = 60
